### PR TITLE
doc: add guidelines for contributing

### PR DESCRIPTION
**Pull request by piscisaureus**
Add contribution guidelines. It's adapted from the one we had in node, but includes code style rules. It also explicitly 
states that we want to be gender/race/everything inclusive so we can avoid #1015 in the future.

Suggested reviewers: @bnoordhuis @isaacs

### CODE REVIEW

**Comment by rsslldnphy** [comment]
I know I'm not really adding anything to the conversation with this, so apologies for that, but: :heart:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+$ git checkout -b my-feature-branch -t origin/v0.10
+```
+
+(Where v0.10 is the latest stable branch as of this writing.)
+
+### CODE
+
+Please adhere to libuv's code style. In general it follows the conventions from
+the [Google C/C++ style guide]. Some of the key points, as well as some
+additional guidelines, are iterated below.
```
Comment (bnoordhuis) [inline CONTRIBUTING.md]: enumerated?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+* Code that is specific to unix-y platforms should be places in `src/unix`, and
```
Comment (bnoordhuis) [inline CONTRIBUTING.md]: placed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  declarations go into `src/uv-unix.h`.
+
+* Source code that is windows-specific goes into `src/win`, and related
```
Comment (bnoordhuis) [inline CONTRIBUTING.md]: Windows-specific?

**Comment by bnoordhuis** [comment]
Some nits but otherwise LGTM.

**Comment by bnoordhuis** [comment]
Landed in 39db225.

**Pull request closed without merging**